### PR TITLE
feat: configurable keymaps

### PR DESCRIPTION
**Pull request by Ramilito**
This is working well and allows the user to override any functions they want.

Remaining is to update the hints with the correct key when overridden (should be easy to lookup) and to update readme

- [x] Add plugin prefix to our plug functions
- [ ] Update readme
- [x] Dynamic hints

### CODE REVIEW

**Comment by Ramilito** [comment]
@mosheavni  I was thinking more along these lines instead of creating our own DSL

**Comment by mosheavni** [comment]
> @mosheavni I was thinking more along these lines instead of creating our own DSL

how so?

**Comment by Ramilito** [comment]
> > @mosheavni I was thinking more along these lines instead of creating our own DSL
> 
> how?

`<Plug>` is a vim keyword, the views would just expose a function and not bother with who triggers the function.
We would then see if that function is mapped to anything and if it is not we set the default ourselves.


The user would override the keymap like this example below:
2024-08-28T12-27-18Z_code

File: lua/kubectl/views/clusterrolebinding/definition.lua [hunk 1]
```diff
@@ -5,7 +5,7 @@ local M = {
   ft = "k8s_clusterrolebinding",
   url = { "{{BASE}}/apis/rbac.authorization.k8s.io/v1/clusterrolebindings?pretty=false" },
   hints = {
-    { key = "<gd>", desc = "describe", long_desc = "Describe selected clusterrolebinding" },
+    { key = "<Plug>(kubectl.describe)", desc = "describe", long_desc = "Describe selected clusterrolebinding" },
```
Comment (mosheavni) [inline lua/kubectl/views/clusterrolebinding/definition.lua:8]: do we really need to list it here?
it's part of the global maps and I don't see why see it in the header

Comment (mosheavni) [reply lua/kubectl/views/clusterrolebinding/definition.lua:8]: i mean every definition that defines a global hint (such as describe)

Comment (Ramilito) [reply lua/kubectl/views/clusterrolebinding/definition.lua:8]: yeah I'll move it to the global hints window!

Comment (Ramilito) [reply lua/kubectl/views/clusterrolebinding/definition.lua]: also moved the g? from every ftplugin, made it work on floats or catch errors

**Comment by Ramilito** [comment]
I'll merge it so you can use it, will add the leftover to "update readme" issue

**Pull request merged**